### PR TITLE
Fix #7036: Handler methods should be able to bind to the molgenis tok…

### LINKED ISSUE #7036: Handler methods should be able to bind to the molgenis token as a method parameter
So
```
@GetMapping(URI)
public String init(@TokenParam String token) {
...
}
```

### PR DESCRIPTION
**Pull request by fdlk**
…en as a method parameter

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits

### CODE REVIEW

File: molgenis-security/src/main/java/org/molgenis/security/token/TokenParam.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * @GetMapping("/blah")
+ * public void getBlah(@TokenParam String token){
+ * ...
+ * }
+ * </pre>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface TokenParam
```
Comment (dennishendriksen) [inline molgenis-security/src/main/java/org/molgenis/security/token/TokenParam.java]: Add documentation why this annotation is preferable above Spring data binder.

File: molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java [hunk 2]
```diff
@@ -798,14 +798,8 @@ private boolean isUser2fa()
 
 	@PostMapping("/logout")
 	@ResponseStatus(OK)
-	public void logout(HttpServletRequest request)
+	public void logout(@TokenParam(required = true) String token, HttpServletRequest request)
```
Comment (dennishendriksen) [inline molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java]: ```POST http://localhost:8080/api/v1/logout``` with header ```x-molgenis-token``` <myToken> results in token being ```null```.

Comment (fdlk) [reply molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java]: Not good enough apperantly to make this a @Component.

**Comment by dennishendriksen** [comment]
@fdlk approved, reviewed and tested. can be merged by you or someone else once Travis goes green.

**Pull request merged**